src/js/tabs/firmware_flasher.js

```diff
@@ -797,9 +797,23 @@ firmware_flasher.initialize = function (callback) {
 
                 function onFinish() {
                     const board = FC.CONFIG.boardName;
+                    const boardSelect = $('select[name="board"]');
+                    const boardSelectOptions = $('select[name="board"] option');
+                    const target = boardSelect.val();
+                    let targetAvailable = false;
+
                     if (board) {
-                        $('select[name="board"]').val(board).trigger('change');
-                        GUI.log(i18n.getMessage('firmwareFlasherBoardVerificationSuccess', {boardName: board}));
+                        boardSelectOptions.filter((_, e) => {
+                            if ($(e).text() === board) {
+                                targetAvailable = true;
+                            }
+                        });
```

Comment:
Not sure what's the point of using filter here, this is just iteration so maybe `forEach` or even better `some` if that's available is a better option here.

Reply:
Yeah, filter is obvious not working here. Wil try with some instead

Reply:
Fixed